### PR TITLE
add resource aware completion for core linkerd cmd

### PR DESCRIPTION
**Pull request by dadjeibaah**
Signed-off-by: Dennis Adjei-Baah <dennis@buoyant.io>

### CODE REVIEW

File: pkg/cmd/cmd.go [hunk 1]
```diff
@@ -95,3 +96,32 @@ func ConfigureOutputFlagCompletion(cmd *cobra.Command) {
 			return []string{"basic", "json", "short", "table"}, cobra.ShellCompDirectiveDefault
 		})
 }
+
+// ConfigureKubeContextFlagCompletion sets up resource-aware completion for command
+// flags based off of a kubeconfig
+func ConfigureKubeContextFlagCompletion(cmd *cobra.Command, kubeconfigPath string) {
```
Comment (mateiidavid) [inline pkg/cmd/cmd.go:102]: this is 🔥 🔥  can't wait to use it!

**Review by mateiidavid — APPROVED** [review]
🚀

File: cli/cmd/identity.go [hunk 2]
```diff
@@ -47,16 +47,38 @@ func newCmdIdentity() *cobra.Command {
...
-		Example: ` 
+		Example: `
  # Get certificate from pod foo-bar in the default namespace.
  linkerd identity foo-bar
-		
+
  # Get certificate from all pods with the label name=nginx
  linkerd identity -l name=nginx
 		`,
+		ValidArgsFunction: func(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
```
Comment (Pothulapati) [inline cli/cmd/identity.go:60]: This is so cool, and worked flawlessly! Especially that it takes into account the `--namespace` flag and filters the suggestions!

**Review by Pothulapati — APPROVED** [review]
LGTM :shipit:

**Pull request merged**